src/framework/workspace/internal/workspaceactioncontroller.cpp

```diff
@@ -23,23 +23,35 @@
 
 #include "types/val.h"
 
+#include "log.h"
+
 using namespace muse::workspace;
 using namespace muse::actions;
 
 void WorkspaceActionController::init()
 {
     dispatcher()->reg(this, "select-workspace", this, &WorkspaceActionController::selectWorkspace);
     dispatcher()->reg(this, "configure-workspaces", this, &WorkspaceActionController::openConfigureWorkspacesDialog);
+    dispatcher()->reg(this, "create-workspace", this, &WorkspaceActionController::createNewWorkspace);
+}
+
+void WorkspaceActionController::prepareCurrentWorkspaceForChange()
+{
+    manager()->prepareCurrentWorkspaceForChange();
 }
 
 void WorkspaceActionController::selectWorkspace(const ActionData& args)
 {
+    prepareCurrentWorkspaceForChange();
+
     std::string selectedWorkspace = !args.empty() ? args.arg<std::string>(0) : "";
     setCurrentWorkspaceName(selectedWorkspace);
```

Comment:
It seems like all this should be in the manager... (manager->selectWorkspace(..)) a controller itself usually doesn't do anything, it only knows who to assign it to. The same with other methods. For example, we can add a workspaces management API, and there we would like to simply call methods (from the manager), and not emulate actions by sending actions